### DISCUSSION TITLE
Add "offline" mode

### DISCUSSION THREAD

**Discussion by itsutsu** [ppy/osu#27253] [Ideas]
Many players prefer to play offline as to not submit scores. The only way to do this in lazer is to log out, but then you lose access to convenient features like osu!direct and chat. Considering a middle ground for such players would be appreciated.

**Comment by Stoppedpuma** [comment]
Has already been mentioned a handful of times. #24149, #19912, etc.. The most likely way I can see this being a thing is with the upcoming support of "Practice Mode". If this is an absolute must for you then the best I can recommend for now is logging out of the client and just using the website for the things you mentioned or just using stable with score v2.

**Comment by cl8n** [comment]
this would be nice to have for any reason, but to add a concrete use-case, it would be great for practicing mappools of tournaments without giving your future opponent any info.

**Comment by shiumano** [comment]
If you build from source code yourself, you can reference online for debugging convenience, but cannot submit for fairness.
Well, it would be most desirable to add this feature officially, but this approach currently exists
